### ISSUE TITLE
Not receiving notifications 

### ISSUE THREAD

**Issue by MaggiBA** [Kemmey/Kemmey-TeslaWatch-Public#286]
I’m new to Tesla and, of course, this app. The app appears to have downloaded, allowed login, etc. easily. (What does the alternately flashing “online/loading” line atop the app mean?)

How long should my Model Y be at the location I have designated and left unplugged before I receive a notification? 

I have left it unplugged for a couple of hours, but received no notification.

I checked the “show debug” log and did see under Authentication Events that there were several failed attempts and finally “gave up”. 

I’m using an Apple Watch 6 and my husband (who is having the same issue) is using an AW 2. Both are using the newest iOS possible.

![F47F47E4-196B-4CF5-8B1F-836EC519DDFC](https://user-images.githubusercontent.com/78923229/107674482-02d5e100-6c65-11eb-9555-be957c9f9e6b.png)

**Comment by Kemmey** [comment]
Hey @MaggiBA 

The loading/online title corresponds to the vehicle and app state. So loading/waking will show when the app is trying to load data from the vehicle. Online indicates the app was able to communicate with the vehicle recently. Offline is shown when vehicle was not responding to requests. 

The official Tesla app can talk to the vehicle both via bluetooth if within range, or via the internet. 3rd parties do not have the luxury of bluetooth, so we have to rely on internet connectivity - which by nature - require both watch/phone and vehicle to be online in order to work. That's not usually an issue, but it's important to convey to the user what state the app and vehicle is in, so you'll know if the data on screen can be trusted. So the number in parens indicates how old the data shown is - in normal operation that should be between 0 and 6 seconds - that's when the app is in the foreground. 

When the app is in the background, it will periodically start up and get data from the vehicle - if enabled in settings. The plugged in alert is dependent on this background operation - so the relevant log to check is actually the background events log. And verify that background updates and wake is allowed. Also - and this is probably a bug, come to think of it, the plugged in alerts will only trigger if you've added one of the apps complications to your watch face - because currently, in order to save battery on the watch, if the complication is not in use, the background updates will not trigger. I'll fix that, so they will also trigger if any of the reminders are enabled!

**Comment by Kemmey** [comment]
Just confirmed the background events bug. Yeah - if the apps complication is not added to your watch face, the notifications will not trigger. I've fixed that now - will be included in next update! Thanks for reporting and helping make the app better! :-D

**Comment by Kemmey** [comment]
Oh - and one last gotcha - if vehicle is below 20% charge, no background events will trigger at all - this is to ensure battery life of the vehicle takes priority.

**Comment by MaggiBA** [comment]
Thank you! Very nice explanation.
I’ll add a complication to some of my watch faces. 
Does the watch face need to be the “active” one for the reminders to come thru?

**Comment by Kemmey** [comment]
Yeah - has to be on the active watch face - logic is, if you're not able to see it, there's no reason to spend battery to refresh it.

**Comment by Kemmey** [comment]
(again - this is fixed in the build I'm submitting to the store tonight)

**Comment by MaggiBA** [comment]
Back again.
I still have not received a plug in reminder. (Neither has my husband.) We’ve deliberately left Blaise unplugged to see if we would get a reminder and haven’t received one.
We receive “open” notifications all the time... sometimes causing confusion, but we’ve worked it out between us.
We tried to force a plug in reminder with and without the “add current vehicle location”, even though we have don’t know if the two are related... what is the purpose of adding the location?
I’m using a watch face with a complication. My husband is not. 
We checked the background settings on the watches, iPhones, and Tesla app.  All appear to be on. Oddly enough, the “Watch for Tesla” Settings appear on my phone (iPhone 11), but not my husband’s (iPhone 12 Pro). (He has since deleted the app entirely in an effort to see if that’s the issue.)
There are no background events in the debug log, but there are authentication event errors (debug log sent via email link in app).
What else can I try?
I love the new Siri Shortcuts. Do I assume correctly that if my car is not equipped to handle a shortcut it will not appear on the phone (e.g. close frunk)? Why am I not getting the enable/doable Sentry, the open/close charge port, and some others. 
Thanks again!

**Comment by Kemmey** [comment]
Hey @MaggiBA 

So sorry for the delayed response!

The location is crucial for notifications - so you have to add your home and work etc. to the locations list - it will only send notifications when vehicle is add a marked location - if not, it would notifify you whenever you went shopping etc. 

The logs you sent were from your phone - as that's the only version of the app that supports that feature - and your phone clearly doesn't perform any background activity. I certainly hope this is not the case for your watch - if you check the background events log on your watches, you should see a lot of activity, as you're properly getting the open features notifications (I still need to adjust the frequency of those). 

The plugin notifications will only trigger once per detected location and disconnect - so basically, if car is at your home, you unplug, then you won't get notified. But if vehicle has left your home (or other marked location) and the watch has received data from the vehicle while at a different location - and vehicle then returns to a marked location - and is left unplugged - that will trigger a notification. So if you only drive a short drive - less than half a mile - it will not reset and will not notify. Similarly, if you drive but the app never refreshes data from the vehicle while away - it will not trigger, as the app didn't notice you ever left. Finally, if car is below 20% charge, in order to preserve battery, all background activity will be stopped until the app is manually launched after the battery is above 20%. 

So basically, add your home to the list of monitored locations - go for a mile+ drive - open up the app while away, ensuring the app notices you've left - ensure vehicle is above 20% - return home and leave unplugged - if you open the app while driving, it will schedule a background update every 15 mins while driving is detected, so within 15 minutes of returning home, a notification SHOULD trigger, if left unplugged. 

About shortcuts - note that the shortcuts section in the app itself, was me trying to add sort of a shortcuts gallery of things you could build with the shortcuts actions - not listing the available actions. In the end, I didn't come up with any good ones to ship, so just listed a few of the actions. In the next version this will be much improved, and will list all available actions, including buttons to add desired voice trigger directly in the app to train Siri. 

So for now, in order to see all available actions - and they will all be available, even though your car may not support them - open up the Apple Shortcuts app, add a new shortcut and search for Tesla when adding a new action - you should then see all the actions from my app :-D

Cheers, 

Kim

**Comment by MaggiBA** [comment]
Thanks Kim!
I followed your directions and received a plug in reminder! Again nice explanation.
If you’d like to write the owner’s manual, I’d be happy to flex my editing muscles and help you out.
All the best,
MBAW
